packages/smithy-core/src/smithy_core/interceptors.py

```diff
@@ -148,7 +148,6 @@ def read_before_execution(
         the `response`. If multiple `read_before_execution` methods throw exceptions,
         the latest will be used and earlier ones will be logged and dropped.
         """
-        pass
```

Comment:
Should these be raising NotImplementedError rather than just passing? Having the functions be no-ops seems odd.

Reply:
They very specifically need to be no-ops. The usage is something like

```python
for interceptor in interceptors:
    interceptor.read_before_execution(...)
```

But not every implementation will implement all of these hooks. So if we had them throwing not implemented then we'd have to catch and ignore that, adding a bunch of useless thrashing

Reply:
Should this explicitly be `return None` then as a default if we're not expecting all of these to be implemented? Having a completely bare function is abnormal.

Reply:
Apparently this has already been a hotly debated topic that ruff has taken an opinionated stance on. It feels weird, but we can go with it if people feel strongly about it.

Reply:
Yeah iirc there are some that prefer `pass` some that prefer `...` and so on. Functions have to have some sort of statement to be valid, but that statement can be the doc string. I really don't care so long as it doesn't impact performance.